hack/deploy-karmada-agent.sh

```diff
@@ -4,24 +4,66 @@ set -o errexit
 set -o nounset
 
 REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
-KARMADA_APISERVER_KUBECONFIG=${KARMADA_APISERVER_KUBECONFIG:-"/var/run/karmada/karmada-apiserver.config"}
 
-# The host cluster name which used to install karmada control plane components.
-MEMBER_CLUSTER_NAME=${MEMBER_CLUSTER_NAME:-"member3"}
-MEMBER_CLUSTER_KUBECONFIG=${MEMBER_CLUSTER_KUBECONFIG:-"${HOME}/.kube/member3.config"}
-
-source ${REPO_ROOT}/hack/util.sh
 function usage() {
   echo "This script will deploy karmada agent to a cluster."
-  echo "Usage: hack/deploy-karmada-agent.sh"
-  echo "Example: hack/deploy-karmada.sh"
+  echo "Usage: hack/deploy-karmada-agent.sh <KARMADA_APISERVER_KUBECONFIG> <KARMADA_APISERVER_CONTEXT_NAME> <MEMBER_CLUSTER_KUBECONFIG> <MEMBER_CLUSTER_CONTEXT_NAME>"
+  echo "Example: hack/deploy-karmada-agent.sh ~/.kube/karmada.config karmada-apiserver ~/.kube/karmada.config member1"
 }
 
-export REGISTRY="swr.ap-southeast-1.myhuaweicloud.com/karmada"
-export VERSION="latest"
-kind load docker-image "${REGISTRY}/karmada-agent:${VERSION}" --name="${MEMBER_CLUSTER_NAME}"
+if [[ $# -ne 4 ]]; then
+  usage
+  exit 1
+fi
+
+# check kube config file existence
+if [[ ! -f "${1}" ]]; then
+  echo -e "ERROR: failed to get kubernetes config file: '${1}', not existed.\n"
+  usage
+  exit 1
+fi
+KARMADA_APISERVER_KUBECONFIG=$1
+
+# check context existence
+if ! kubectl config get-contexts "${2}" --kubeconfig="${KARMADA_APISERVER_KUBECONFIG}" > /dev/null 2>&1;
+then
+  echo -e "ERROR: failed to get context: '${2}' not in ${KARMADA_APISERVER_KUBECONFIG}. \n"
+  usage
+  exit 1
+fi
+KARMADA_APISERVER_CLUSTER_NAME=$2
```

Comment:
You name it as KARMADA_APISERVER_CONTEXT_NAME in usage function，but call it KARMADA_APISERVER_CLUSTER_NAME here, it may be a little confused.

Reply:
I notice that you introduce the KARMADA_APISERVER_CONTEXT_NAME param but not used. I think we can combine it with the --karmada-context flag here.  https://github.com/karmada-io/karmada/blob/master/artifacts/agent/karmada-agent.yaml#L29  
For example, sed -i "s/{{karmada_context}}/${KARMADA_APISERVER_KUBECONFIG}/g"

Reply:
let me check it

Reply:
> I notice that you introduce the KARMADA_APISERVER_CONTEXT_NAME param but not used. I think we can combine it with the --karmada-context flag here. https://github.com/karmada-io/karmada/blob/master/artifacts/agent/karmada-agent.yaml#L29
> For example, sed -i "s/{{karmada_context}}/${KARMADA_APISERVER_KUBECONFIG}/g"

That's a good idea

Reply:
> You name it as KARMADA_APISERVER_CONTEXT_NAME in usage function，but call it KARMADA_APISERVER_CLUSTER_NAME here, it may be a little confused.

They are the same purpose and I finally use 'KARMADA_APISERVER_CONTEXT_NAME'